git_machete/cli.py

```diff
@@ -75,38 +78,172 @@ def usage(command: str = None) -> None:
+    anno_parser.add_argument('annotation_text', nargs='*')
+    anno_parser.add_argument('-b', '--branch', default=argparse.SUPPRESS)
+    anno_parser.add_argument(
+        '-H', '--sync-github-prs', action='store_true', default=argparse.SUPPRESS)
+
+    delete_unmanaged_parser = subparsers.add_parser('delete-unmanaged')
+    delete_unmanaged_parser.add_argument(
+        '-y', '--yes', action='store_true', default=argparse.SUPPRESS)
+
+    diff_full_parser = subparsers.add_parser('diff', aliases=['d'])
+    diff_full_parser.add_argument('branch', nargs='?')
+    diff_full_parser.add_argument(
+        '-s', '--stat', action='store_true', default=argparse.SUPPRESS)
+
+    discover_parser = subparsers.add_parser('discover', argument_default=argparse.SUPPRESS)
+    discover_parser.add_argument('-C', '--check-out-since')
+    discover_parser.add_argument('-l', '--list-commits', action='store_true')
+    discover_parser.add_argument('-r', '--roots')
+    discover_parser.add_argument('-y', '--yes', action='store_true')
+
+    subparsers.add_parser('edit', aliases=['e'])
+
+    subparsers.add_parser('file')
+
+    fork_point_parser = subparsers.add_parser('fork-point', argument_default=argparse.SUPPRESS)
+    fork_point_parser.add_argument('branch', nargs='?', default=None)
+    fork_point_parser.add_argument('--inferred', action='store_true')
+    fork_point_parser.add_argument('--override-to')
+    fork_point_parser.add_argument('--override-to-inferred', action='store_true')
+    fork_point_parser.add_argument('--override-to-parent', action='store_true')
+    fork_point_parser.add_argument('--unset-override', action='store_true')
+
+    subparsers.add_parser('format')
+
+    github_parser = subparsers.add_parser('github')
+    github_parser.add_argument('subcommand', choices=['anno-prs', 'create-pr', 'retarget-pr'])
+    github_parser.add_argument('--draft', action='store_true', default=argparse.SUPPRESS)
+
+    go_parser = subparsers.add_parser('go', aliases=['g'])
+    go_parser.add_argument(
+        'direction',
+        choices=['d', 'down', 'f', 'first', 'l', 'last', 'n', 'next',
+                 'p', 'prev', 'r', 'root', 'u', 'up']
+    )
+
+    help_parser = subparsers.add_parser('help')
+    help_parser.add_argument(
+        '--command', choices=list(short_docs.keys()), default=argparse.SUPPRESS)
+
+    subparsers.add_parser('hooks')
+
+    is_managed_parser = subparsers.add_parser('is-managed')
+    is_managed_parser.add_argument('branch', nargs='?')
+
+    list_parser = subparsers.add_parser('list')
+    list_parser.add_argument(
+        'category',
+        choices=[
+            'addable', 'managed', 'slidable', 'slidable-after', 'unmanaged',
+            'with-overridden-fork-point']
+    )
+    list_parser.add_argument('branch', nargs='?', default=argparse.SUPPRESS)
+
+    log_parser = subparsers.add_parser('log', aliases=['l'])
+    log_parser.add_argument('branch', nargs='?', default=argparse.SUPPRESS)
```

Comment:
Hmmm what does `nargs='?'` mean? is it "0 or more arguments"? or "0 or 1 argument"?

Reply:
0 or 1, but if there is 0 use default. https://docs.python.org/3/library/argparse.html#nargs

Reply:
Ok, looks that `+`, `?` and  `*` are used consistently with regex syntax here